crates/ty_python_semantic/src/types/overrides.rs

```diff
@@ -335,6 +361,43 @@ fn check_class_declaration<'db>(
     }
 }
 
+/// Check for prohibited `NamedTuple` attribute overrides.
+///
+/// `NamedTuple` classes have certain synthesized attributes (like `_asdict`, `_make`, etc.)
+/// that cannot be overwritten. Attempting to assign to these attributes (without type
+/// annotations) or define methods with these names will raise an `AttributeError` at runtime.
+fn check_prohibited_namedtuple_attrs<'db>(
+    context: &InferContext<'db, '_>,
+    own_class_members: &FxHashSet<MemberWithDefinition<'db>>,
+) {
+    let db = context.db();
+    let module = context.module();
+
+    for member in own_class_members {
+        if !PROHIBITED_NAMEDTUPLE_ATTRS.contains(&member.member.name.as_str()) {
+            continue;
+        }
+
+        let Some(definition) = member.definition else {
+            continue;
+        };
+
+        // Only annotated assignments are allowed (they become fields).
+        // Everything else (assignments, method definitions, etc.) is an error.
+        if !matches!(definition.kind(db), DefinitionKind::AnnotatedAssignment(_)) {
+            if let Some(builder) = context.report_lint(
+                &OVERRIDE_OF_PROHIBITED_NAMED_TUPLE_ATTRIBUTE,
+                definition.full_range(db, module),
```

Comment:
we should use `.focus_range()` here rather than `.full_range()` -- the full range might be huge!